src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -175,7 +175,15 @@ private static void setModules(TestInputConfiguration.Builder testInputConfigBui
             do {
                 lineNo++;
             } while (lines.get(lineNo).isEmpty() || !lines.get(lineNo - 1).isEmpty());
-        } while (!lines.get(lineNo).startsWith("*/"));
+        } while (!startsWithClosingComment(lines.get(lineNo)));
+    }
+
+    private static boolean startsWithOpeningComment(String line) {
+        return line.startsWith("/*") || line.startsWith("<!--");
```

Comment:
Please check https://github.com/checkstyle/checkstyle/pull/13219#discussion_r1228944093.
> While the html config "comment" is debatable (I am not a big fan), this is not valid xml. As we demand compilation for java files, we should have valid xml. Are we able to simply put this comment inside of a valid multiline xml comment? If fix is that easy, we should also update html file.

We want to have it for inline configs in `xml`, `html`. Migration to `/*xml` is a separate issue.

Reply:
Ok, it might be better to limit usage of not java files as much as possible, and keep them as exception for now and deal with them at the end of project. 
I am ok to have such Examples to out of our project at all. 
Let's focus on java problem first.